fastlane/lanes/release.rb

```diff
@@ -203,36 +312,43 @@ def report_milestone_error(error_title:)
   buildkite_annotate(style: 'warning', context: 'error-with-milestone', message: error_message) if is_ci
 end
 
-def create_release_management_pull_request(release_version:, base_branch:, title:)
-  # TODO: Adopt shared EnvManager ASAP. See https://github.com/wordpress-mobile/release-toolkit/pull/578
-  # token = EnvManager.get_required_env!('GITHUB_TOKEN')
-  token = ENV.fetch('GITHUB_TOKEN', nil)
-  UI.user_error!('No GITHUB_TOKEN found in the environment.') if token.nil?
-
-  pr_url = create_pull_request(
-    api_token: token,
-    repo: GITHUB_REPO,
-    title: title,
-    head: Fastlane::Helper::GitHelper.current_git_branch,
-    base: base_branch,
-    labels: 'Releases'
-  )
+def create_backmerge_pr!
```

Comment:
What do you think of what I did here?

- `create_backmerge_pr!` does exactly what is says, returns the URL of the PR or crashes if something went wrong. This allows callers to declare they expect a single backmerge PR, like in this repo for a code freeze or a release finalization.
- `create_backmerge_prs!` below allows for more backmerge PRs, like when we finalize a hotfix. It also has the `!` suffix because if something goes wrong it will call `UI.user_error!`

Reply:
I wrote a [short comment about this in another PR](https://github.com/Automattic/simplenote-ios/pull/1655#discussion_r1755228424) but I guess it got lost amidst all PRs and discussions currently happening.

My main question is whether we can be sure the validation on `create_backmerge_pr!` is guaranteed to be semantically correct, meaning: I think there may be cases when a previous release might take a bit longer or there is a hotfix in progress, then you "unexpectedly" will have a backmerge.

Reply:
I see... 🤔 

So, for example if `finalize_code_freeze` for `1.2` runs when `1.1` hasn't landed yet? Or if we are code freezing `1.2` while `1.1.1` is still in progress?

I'm tempted to say that 1) that should not happen because one should not run a code freeze before having finalized the previous release and 2) [the action should only create backmerges in branches that are from later releases](https://github.com/wordpress-mobile/release-toolkit/blob/ee03eb9232597fd812c18d3fb0fbf91130facf97/spec/create_release_backmerge_pull_request_spec.rb#L97-L116) (e.g. 1.2.1 or 1.3). That to me suggests that if we have more than one backmerge then something went wrong and we should abort.

Having said that... Maybe it's best to be flexible (and account for me not being able to predict every future possible case in which this action should run) and let release manager deal with multiple backmerge PRs when only one should be there.

In fact, now that I think about it, the release manager would have to deal with them anyway because the failure implemented here is _after_ the PRs have been created.

Here's what I might do:

- Remove the `create_backmerge_pr!`
- Move the logic to print to STDOUT annotate CI into `create_backmerge_prs!` (I am keen on keeping the plural form to make it clear at call site that more than one might be opened)
- Add a `warn_on_multiple_prs_opened: false` parameter that lanes such as `finalize_code_freeze` can use to change the STDOUT and CI annotation to include a warning if more than one backmerge was create

What do you think @iangmaia ?

Reply:
Actually @iangmaia sorry I'm going to merge this before you have a chance to reply to this because I had to make a few more changes to the release branch and will need to submit a new build. As such, it's convenient to have this PR on `trunk` before the `publish_release` task will open a new integration PR (because of the new changes).

Reply:
> So, for example if finalize_code_freeze for 1.2 runs when 1.1 hasn't landed yet? Or if we are code freezing 1.2 while 1.1.1 is still in progress?

Yeah, in that case the PR would be created from 1.1.1 to 1.2 on `finalize_hotfix`, but I guess it's just a matter of using the plural method in this case.

> In fact, now that I think about it, the release manager would have to deal with them anyway because the failure implemented here is after the PRs have been created.

Makes sense to me 👍  I'm still a bit on the fence on what's best to do but annotating a warning sounds like a good compromise.

> sorry I'm going to merge this before you have a chance to reply to this because I had to make a few more changes to the release branch and will need to submit a new build

No worries!

Reply:
A minor followup: 

> Move the logic to print to STDOUT annotate CI into create_backmerge_prs! (I am keen on keeping the plural form to make it clear at call site that more than one might be opened)

I run into the case of using `create_backmerge_pr` expecting it to return a single URL in[ this Simplenote iOS build](https://buildkite.com/automattic/simplenote-ios/builds/1160#01921de0-8921-4b6c-93cf-53dd88a882fd). There result was a buggy UX where the link to the PR in the build annotation didn't work. The reason? The markup had an array in the URL with got rendered as empty:

![image](https://github.com/user-attachments/assets/8854c6da-2829-47e6-8429-7fbbf0003a2e)

Reply:
Following up via https://github.com/Automattic/simplenote-android/pull/1698